packages/rln/src/credentials_manager.ts

```diff
@@ -0,0 +1,306 @@
+import { hmac } from "@noble/hashes/hmac";
+import { sha256 } from "@noble/hashes/sha256";
+import { Logger } from "@waku/utils";
+import { ethers } from "ethers";
+
+import { LINEA_CONTRACT } from "./contract/constants.js";
+import { RLNBaseContract } from "./contract/rln_base_contract.js";
+import { IdentityCredential } from "./identity.js";
+import { Keystore } from "./keystore/index.js";
+import type {
+  DecryptedCredentials,
+  EncryptedCredentials
+} from "./keystore/index.js";
+import { KeystoreEntity, Password } from "./keystore/types.js";
+import { RegisterMembershipOptions, StartRLNOptions } from "./types.js";
+import {
+  buildBigIntFromUint8Array,
+  extractMetaMaskSigner
+} from "./utils/index.js";
+import { Zerokit } from "./zerokit.js";
+
+const log = new Logger("waku:credentials");
+
+/**
+ * Manages credentials for RLN
+ * This is a lightweight implementation of the RLN contract that doesn't require Zerokit
+ * It is used to register membership and generate identity credentials
+ */
+export class RLNCredentialsManager {
+  protected started = false;
+  protected starting = false;
+
+  private _contract: undefined | RLNBaseContract;
+  private _signer: undefined | ethers.Signer;
+
+  protected keystore = Keystore.create();
+  private _credentials: undefined | DecryptedCredentials;
+
+  public zerokit: undefined | Zerokit;
+
+  public constructor(zerokit?: Zerokit) {
+    log.info("RLNCredentialsManager initialized");
+    this.zerokit = zerokit;
+  }
+
+  public get contract(): undefined | RLNBaseContract {
```

Comment:
I am not sure if it was the case before, but these getters don't really do anything 
Let's just make `contract`, `signer` and `credentials` public.